packages/eslint-plugin/rules/a11y-anchor-has-content.js

```diff
@@ -0,0 +1,95 @@
+/**
+ *
+ * @file Wraps eslint-plugin-jsx-a11y/anchor-has-content to allow usage with createInterpolateElement.
+ *
+ */
+
+/**
+ * External dependencies
+ *
+ * eslint-plugin-jsx-a11y/anchor-has-content: to create a wrapper around it, we need to use it
+ * elementType from jsx-ast-utils: to get the type of the node we are working with.
+ */
+const anchorRule = require( 'eslint-plugin-jsx-a11y' ).rules[
+	'anchor-has-content'
+];
+/**
+ * Internal dependencies
+ */
+const { getTextContentFromNode } = require( '../utils' );
+const meta = {
+	type: 'problem',
+	docs: {
+		description:
+			'Checks that anchors are not empty and have an href, relies on eslint-plugin-jsx-a11y/anchor-has-content under the hood. Supports usage of createInterpolateElement.',
+	},
+	fixable: 'code',
+	messages: {
+		anchorHasContent:
+			'Anchors must have content and the content must be accessible by a screen reader. Check the first parameter to createInterpolateElement.',
+		invalidMarkup:
+			'The first parameter to createInterpolateElement does not contain valid markup for an anchor.',
+	},
+	schema: [], // no options
+};
+
+const getATagsContent = ( str ) => {
+	const tagParts = [ ...str.matchAll( /<a>(.*?)<\/a>/g ) ];
+	return tagParts.map( ( element ) => {
+		return element[ 1 ].trim();
+	} );
+};
+
+const rule = function ( context ) {
+	return {
+		JSXOpeningElement: ( node ) => {
+			const isAnchor = node?.name?.name === 'a';
+			if ( ! isAnchor ) {
+				// not an anchor, so we don't care
+				return;
+			}
+			const interpolatedEl = context
+				.getAncestors()
```

Comment:
How deep does this go in the tree? Can it potentially catch false positives in case there's a nested combination of `createInterpolateElement` and anchors?

Reply:
It goes up to "Program" but there are no false positives as far as I can see. We only get one through `find`, and because the parent of an <a> will never be another <a>, and we don't navigate recursively through all the callees of each parent, I _think_ we are safe. 
See [this test](https://github.com/WordPress/gutenberg/pull/50065/files#diff-4c555e1042f88bdebf5cba43ba582ae78b721c7d64c1d0cc527ea1975ef191edR269-R278), where I tried to demonstrate the concept with 3 elements. LMK if you think I missed something!

Reply:
Got it, thanks for double-checking and adding tests for the case!

> and because the parent of an will never be another

Per docs, it seems that `getAncestors` will actually return them in order, starting from the direct parent, so this should be good even if it returns all other entires (as `find` will match the first one, like you said)